packages/jsii-pacmak/test/generated-code/__snapshots__/target-java.test.ts.snap

```diff
@@ -6642,6 +6979,217 @@ public interface DerivedStruct extends software.amazon.jsii.JsiiSerializable, so
 
 `;
 
+exports[`Generated code for "jsii-calc": <outDir>/java/src/main/java/software/amazon/jsii/tests/calculator/DiamondBottom.java 1`] = `
+package software.amazon.jsii.tests.calculator;
+
+/**
+ */
+@javax.annotation.Generated(value = "jsii-pacmak")
+@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.DiamondBottom")
+@software.amazon.jsii.Jsii.Proxy(DiamondBottom.Jsii$Proxy.class)
+@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
+public interface DiamondBottom extends software.amazon.jsii.JsiiSerializable, software.amazon.jsii.tests.calculator.lib.DiamondLeft, software.amazon.jsii.tests.calculator.lib.DiamondRight {
+
+    /**
+     */
+    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
+    default @org.jetbrains.annotations.Nullable java.time.Instant getBottom() {
+        return null;
+    }
+
+    /**
+     */
+    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Deprecated)
+    @Deprecated
+    default @org.jetbrains.annotations.Nullable java.lang.String getHoistedTop() {
+        return null;
+    }
```

Comment:
This right here is what used to not be emitted - leading to `DiamondBottom` inheriting the default implementation from both `DiamondLeft` and `DiamondRight` (both of which hoisted that from their internal parent `InternalDiamondTop`) -- causing illegal dispatch ambiguity.

New default declared here replaces the parent's, and solves the issue.